src/model/element.ts

```diff
@@ -70,6 +72,7 @@ export class Element implements Feature {
   jsdoc?: jsdoc.Annotation;
   astNode: estree.Node|null;
   kinds: Set<string> = new Set(['element']);
+  warnings: Warning[] = [];
```

Comment:
Feature is just an interface so we need to add this onto all implementing classes, but yes. This PR adds an array of warnings onto ~everything, for getWarnings to gather up